src/app/project/board/attribution-tab-view/attribution-tab-view.component.ts

```diff
@@ -8,15 +8,32 @@ import { IAttribution, IOrganisation, IPerson } from '../dataset-metadata';
 })
 export class AttributionTabViewComponent {
     // attribution input
-    @Input() attributions: IAttribution[];
+    @Input() attributions: Attribution[];
+
+    @Input() subProperties: Object;
+
+    currentAgent: Person | Organization | IId;
 
     // return the type of agent to use correct template to display it
-    getAgentType(agent: IPerson | IOrganisation) {
-        if (agent.hasOwnProperty('familyName')) {
+    setAgent (agent: Person | Organization | IId) {
+        let atype = this.getAgentType(agent);
+        if (atype) {
+            this.currentAgent = agent;
+            return atype;
+        }
+        this.currentAgent = this.subProperties[agent.id];
+        atype = this.getAgentType(this.currentAgent);
+        return atype;
```

Comment:
why not `if...else` with one `return` outside?

Reply:
When I used if..else before, linter was complaining about else condition and my tests were failing because of that. So I updated the code like this.

Reply:
Same here, [this answer](https://github.com/dasch-swiss/dsp-app/pull/400#discussion_r585398051).